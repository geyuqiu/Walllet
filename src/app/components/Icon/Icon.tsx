import {SvgCollection} from "app/assets/svg";
import React from "react";
import styled from "styled-components";

type Props = {
	name: string;
	width: number | string;
	height: number | string;
	as?: React.ElementType;
	fallback?: React.ReactNode;
	fill?: string;
	stroke?: string;
} & React.HTMLProps<any>;

type WrapperProps = {
	width: number | string;
	height: number | string;
};

const Wrapper = styled.div(({ width, height }: WrapperProps) => ({
	svg: {
		width,
		height,
	},
}));

export const Icon = ({ name, width, height, fallback, fill, stroke, ...props }: Props) => {
	const Svg = SvgCollection[name];

	return (
		<Wrapper width={width} height={height} {...props}>
			{Svg ? <Svg fill={fill} stroke={stroke}/> : fallback}
		</Wrapper>
	);
};

Icon.defaultProps = {
	width: "1em",
	height: "1em",
};
