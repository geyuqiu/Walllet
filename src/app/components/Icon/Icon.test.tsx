import {render, screen} from "@testing-library/react";
import React from "react";

import {Icon} from "./Icon";

describe("Icon", () => {
	it("should render", () => {
		const { container, asFragment } = render(<Icon name="ARK" />);

		expect(container).toBeTruthy();
		expect(asFragment()).toMatchSnapshot();
	});

	it("should render fallback", () => {
		const { asFragment } = render(<Icon name="unknown" fallback={<span>Not found</span>} />);

		expect(screen.getByText("Not found"));

		expect(asFragment()).toMatchSnapshot();
	});
});
