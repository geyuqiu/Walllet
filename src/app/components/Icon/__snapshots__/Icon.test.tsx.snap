// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon should render 1`] = `
<DocumentFragment>
  <div
    class="sc-bdfBwQ fPrsah"
    height="1em"
    width="1em"
  />
</DocumentFragment>
`;

exports[`Icon should render fallback 1`] = `
<DocumentFragment>
  <div
    class="sc-bdfBwQ fPrsah"
    height="1em"
    width="1em"
  >
    <span>
      Not found
    </span>
  </div>
</DocumentFragment>
`;
